### PR TITLE
Implement per-user-key revoke, unskip tests

### PR DESCRIPTION
**Pull request by mlsteele**
Make revoking work with per-user-keys

New version of https://github.com/keybase/client/pull/6853

### CODE REVIEW

File: go/libkb/kbsig.go [hunk 1]
```diff
@@ -648,3 +648,57 @@ func PerUserKeyProof(me *User,
...
+	}
+	sig, sigID, linkID, err := SignJSON(jw, signer)
+	if err != nil {
+		return nil, err
+	}
+
+	// Update the user locally
+	me.SigChainBump(linkID, sigID)
+
+	// TODO the user is now inconsistent. The sigchain has been bumped but the semantics of the link
```
Comment (maxtaco) [inline go/libkb/kbsig.go]: What's up with this `TODO`?

File: go/engine/deprovision_test.go [hunk 2]
```diff
@@ -342,3 +415,32 @@ func TestCurrentDeviceRevoked(t *testing.T) {
...
+	})
+	assertNumDevicesAndKeys(tc, fu, 0, 0)
+
+	// Now, test deprovision codepath with no secret store
+	tc.G.SecretStoreAll = nil
+	fu = assertDeprovisionWithSetup(tc, assertDeprovisionWithSetupArg{
+		revokePaperKey: true,
+	})
+	assertNumDevicesAndKeys(tc, fu, 0, 0)
+}
```
Comment (maxtaco) [inline go/engine/deprovision_test.go]: Is there a test that tests making a device, revoking a device, and then adding a new device that has to follow the prevs, and for that device to use the prev decryption logic to follow back to the previous keys?

**Comment by mlsteele** [comment]
Added local delegation of per-user-keys (second-to-last commit). `PerUserKeyProofReverseSigned` calls `User::localDelegatePerUserKey` which inserts the key into the local CKI.

File: go/engine/deprovision_test.go (hunk 2 above)
Comment (mlsteele) [inline go/engine/deprovision_test.go]: Yes, there is `TestRevokerPaperDeviceTwicePUK` which does this:
- signup (adds puk generation 1)
- make a second paper key
- revoke paper key (adds puk generation 2)
- revoke paper key (adds puk generation 3)
- checks the keyring before and after a sync to make it sure it is on generation 3. It doesn't check the actual key contents.

**Pull request merged**